site/components/src/main/java/uk/nhs/digital/common/components/ApiCatalogueComponent.java

```diff
@@ -0,0 +1,18 @@
+package uk.nhs.digital.common.components;
+
+import org.hippoecm.hst.core.component.HstRequest;
+import org.hippoecm.hst.core.component.HstResponse;
+
+import java.util.Optional;
+
+public class ApiCatalogueComponent extends BaseGaContentComponent {
+
+    @Override public void doBeforeRender(final HstRequest request,
+                                         final HstResponse response) {
+        super.doBeforeRender(request, response);
+
+        Optional.ofNullable(request.getRequestContext().getContentBean()).ifPresent(document -> {
```

Comment:
I'd be tempted to try removing this entire statement and see what happens.

One of the methods in one of the the super classes already invokes `request.setModel` which, I think, achieves the same objective.

Either way, we can keep the current class itself because we'll soon need it when we add support for filtering by taxonomy.